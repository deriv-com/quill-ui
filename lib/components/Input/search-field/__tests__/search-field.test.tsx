import React, { ComponentProps } from "react";
import { render, fireEvent, screen, act } from "@testing-library/react";
import SearchField from "..";

jest.mock("@deriv/quill-icons", () => ({
    ...jest.requireActual("@deriv/quill-icons"),
    StandaloneCircleCheckBoldIcon: jest.fn(() => "mockedSuccessIcon"),
    StandaloneSearchRegularIcon: jest.fn(() => "mockedSearchIcon"),
    StandaloneTriangleExclamationBoldIcon: jest.fn(() => "mockedErrorIcon"),
    StandaloneCircleXmarkFillIcon: jest.fn(() => "mockedClearIcon"),
}));

describe("<SearchField />", () => {
    const renderSearchField = (
        props: ComponentProps<typeof SearchField> = {},
    ) => {
        return render(<SearchField {...props} />);
    };
    it("renders correctly with default props", () => {
        const { container } = renderSearchField();
        expect(screen.getByText("mockedSearchIcon")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("updates value correctly on input change", () => {
        const onChangeMock = jest.fn();
        const { container } = renderSearchField({ onChange: onChangeMock });
        const searchElement = screen.getByPlaceholderText("Search");

        act(() => {
            fireEvent.change(searchElement, { target: { value: "test" } });
        });

        expect(onChangeMock).toHaveBeenCalledTimes(1);
        expect(onChangeMock).toHaveBeenCalledWith(
            expect.objectContaining({
                target: expect.objectContaining({
                    value: "test",
                }),
            }),
        );
        expect(container).toMatchSnapshot();
    });
    it("renders clear icon when value is not empty and clear icon should be removed after click", () => {
        const { container } = renderSearchField({ value: "test" });
        const mockedClearIcon = screen.getByText("mockedClearIcon");
        expect(mockedClearIcon).toBeInTheDocument();

        act(() => fireEvent.click(mockedClearIcon));
        
        expect(mockedClearIcon).not.toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it('renders success status icon when status prop is "success"', () => {
        const { container } = renderSearchField({ status: "success" });
        const mockedSuccessIcon = screen.getByText("mockedSuccessIcon");
        expect(mockedSuccessIcon).toBeInTheDocument();
        expect(mockedSuccessIcon).toHaveClass("status-icon--success");
        expect(container).toMatchSnapshot();
    });
    it('renders error status icon when status prop is "error"', () => {
        const { container } = renderSearchField({ status: "error" });
        const mockedErrorIcon = screen.getByText("mockedErrorIcon");
        expect(mockedErrorIcon).toBeInTheDocument();
        expect(mockedErrorIcon).toHaveClass("status-icon--error");
        expect(container).toMatchSnapshot();
    });
    it('renders leftStatusMessage when prop is given', () => {
      const { container } = renderSearchField({ leftStatusMessage: "mocked status message" });
      const statusMessage = screen.getByText('mocked status message')
      expect(statusMessage).toBeInTheDocument();
      expect(statusMessage).toHaveClass('message__container__text');
      expect(container).toMatchSnapshot();
    })
});
