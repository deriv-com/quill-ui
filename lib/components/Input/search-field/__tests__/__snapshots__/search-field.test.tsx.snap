// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> renders clear icon when value is not empty and clear icon should be removed after click 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--neutral"
    >
      <span
        class="icon_wrapper"
      >
        mockedSearchIcon
      </span>
      <div
        class="quill-input-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          placeholder="Search"
          type="text"
          value="test"
        />
      </div>
    </div>
    <div
      class="message__container"
    />
  </div>
</div>
`;

exports[`<SearchField /> renders correctly with default props 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--neutral"
    >
      <span
        class="icon_wrapper"
      >
        mockedSearchIcon
      </span>
      <div
        class="quill-input-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <div
      class="message__container"
    />
  </div>
</div>
`;

exports[`<SearchField /> renders error status icon when status prop is "error" 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--error"
    >
      <span
        class="icon_wrapper"
      >
        mockedSearchIcon
      </span>
      <div
        class="quill-input-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          placeholder="Search"
          type="text"
        />
      </div>
      <span
        class="icon_wrapper status-icon--error"
      >
        mockedErrorIcon
      </span>
    </div>
    <div
      class="message__container"
    />
  </div>
</div>
`;

exports[`<SearchField /> renders leftStatusMessage when prop is given 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--neutral"
    >
      <span
        class="icon_wrapper"
      >
        mockedSearchIcon
      </span>
      <div
        class="quill-input-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <div
      class="message__container"
    >
      <p
        class="quill-typography__caption-text__weight--regular__decoration--default quill-typography__color--prominent message__container__text message__container__text__status--neutral"
      >
        mocked status message
      </p>
    </div>
  </div>
</div>
`;

exports[`<SearchField /> renders success status icon when status prop is "success" 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--success"
    >
      <span
        class="icon_wrapper"
      >
        mockedSearchIcon
      </span>
      <div
        class="quill-input-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          placeholder="Search"
          type="text"
        />
      </div>
      <span
        class="icon_wrapper status-icon--success"
      >
        mockedSuccessIcon
      </span>
    </div>
    <div
      class="message__container"
    />
  </div>
</div>
`;

exports[`<SearchField /> updates value correctly on input change 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper--has-value quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--neutral"
    >
      <span
        class="icon_wrapper"
      >
        mockedSearchIcon
      </span>
      <div
        class="quill-input-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <div
      class="message__container"
    />
  </div>
</div>
`;
